src/models/piranha_arguments.rs

```diff
@@ -286,6 +292,18 @@ impl PiranhaArgumentsBuilder {
     info!( "Number of rules and edges loaded : {:?}", _arg.rule_graph().get_number_of_rules_and_edges());
     _arg
   }
+
+  fn _validate(&self) -> Result<bool, String> {
+    let _arg = self.create().unwrap();
+    if _arg.code_snippet().is_empty() && _arg.path_to_codebase().is_empty() {
```

Comment:
Aren't we also missing a check here that we don't have a case where both are present? Given that the code in `impl Piranha` above just silently uses the code snippet if both are present, and that's not documented or obvious from the API.

Potentially the "ideal" solution is to somehow take these as repeating arguments and merging them (you can pass N directories and M code snippets and Piranha analyzes them all). But that also doesn't seem worth the implementation complexity. So, my suggestion is to ensure that _exactly_ one of `path_to_codebase` and `code_snippet` was passed (you could make the check above an xor, but I would just do two separate checks for the sake of better error messages 😉 )

Reply:
updated `_validate` function